src/App.tsx

```diff
@@ -1,7 +1,37 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+interface State {
+  key: string | null;
+}
+
+export class App extends React.Component<{}, State> {
+  state = {
+    key: null,
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({ key: event.key });
+    });
+  }
+
+  componentWillUnmount(): void {
+    document.addEventListener('keyup', (event: KeyboardEvent) => {
```

Comment:
here you should remove event listener with "removeEventListener" instead  of add another event listener
2. [CODE KNOWLEDGE] - Check if you remove your event listener, when component will unmount from page.